### PR TITLE
feat: Log shader link errors

### PR DESCRIPTION
**Pull request by janisozaur**
## Summary
Log shader link errors just like compilation ones

### CODE REVIEW

File: source/Shader.cpp [hunk 1]
```diff
@@ -43,8 +43,16 @@ Shader::Shader(const char *vertex, const char *fragment)
 	
 	GLint status;
 	glGetProgramiv(program, GL_LINK_STATUS, &status);
-	if(status == GL_FALSE)
+	if(status == GL_FALSE) {
```
Comment (tehhowch) [inline source/Shader.cpp:46]: Please match the style of the existing code

(braces on newline when following parentheses, no `std::` prefix when not required)

Comment (janisozaur) [reply source/Shader.cpp:46]: What tool do I use to apply your formatting style? I haven't seen any clang-format rules and the reply hardly looks like the default one.

Comment (tehhowch) [reply source/Shader.cpp:46]: There are no automated tool rules for the conventions used by this project. We do have editorconfig support for the whitespace requirements.

Other than that, there is the style guide:  https://endless-sky.github.io/styleguide/styleguide.xml

Comment (janisozaur) [reply source/Shader.cpp:46]: Well, that's quite surprising. How about clang-format then?

Comment (janisozaur) [reply source/Shader.cpp:46]:
```suggestion
	if(status == GL_FALSE)
	{
```

File: source/Shader.cpp [hunk 2]
```diff
@@ -43,8 +43,16 @@ Shader::Shader(const char *vertex, const char *fragment)
...
+		GLint maxLength = 0;
+		glGetProgramiv(program, GL_INFO_LOG_LENGTH, &maxLength);
+		std::vector<GLchar> infoLog(maxLength);
```
Comment (janisozaur) [inline source/Shader.cpp:49]:
```suggestion
		vector<GLchar> infoLog(maxLength);
```

**Pull request merged**